### PR TITLE
fix: unit tests failing in CI

### LINKED ISSUE #2223: Unit tests failing in CI
### Description

At first it seems that unit tests are passing, but they are actually failing with `FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory`

### Steps to reproduce

1. Re-run the tests in the last commit to main

### Additional context

Example run https://github.com/hashgraph/hedera-json-rpc-relay/actions/runs/8347903296/job/22848624989

### Hedera network

other

### Version

main

### Operating system

None

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:

This PR updates the precheck.spec.ts file to resolve a TypeError occurring in the ethers package when using node version 20. The error arises because the 'from' field in the transaction doesn't match the signer's address.
It also adds return types and JSDoc where missing.

**Related issue(s)**:

Fixes #2223 

**Notes for reviewer**:

**Checklist**
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Ivo-Yankov — DISMISSED** [review]

**Review by natanasow — APPROVED** [review]
LGTM :clap:

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Review by AlfredoG87 — DISMISSED** [review]

File: packages/relay/src/lib/precheck.ts [hunk 1]
```diff
@@ -80,9 +106,13 @@ export class Precheck {
...
   /**
-   * @param tx
+   * Checks the nonce of the transaction.
+   * @param {Transaction} tx - The transaction.
+   * @param {number} accountInfoNonce - The nonce of the account.
+   * @param {string} [requestId] - The request ID.
+   * @returns {Promise<void>} A Promise.
    */
-  async nonce(tx: Transaction, accountInfoNonce: number, requestId?: string) {
+  async nonce(tx: Transaction, accountInfoNonce: number, requestId?: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (natanasow) [reply packages/relay/src/lib/precheck.ts:115]: @konstantinabl still can't understand why this method must be `async` :confused:

File: packages/relay/src/lib/precheck.ts [hunk 2]
```diff
@@ -153,10 +187,13 @@ export class Precheck {
...
-   * @param tx
-   * @param callerName
+   * Checks the balance of the sender account.
+   * @param {Transaction} tx - The transaction.
+   * @param {any} account - The account information.
+   * @param {string} [requestId] - The request ID.
+   * @returns {Promise<void>} A Promise.
    */
-  async balance(tx: Transaction, account: any, requestId?: string) {
+  async balance(tx: Transaction, account: any, requestId?: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (natanasow) [reply packages/relay/src/lib/precheck.ts:196]: @konstantinabl

**Review by natanasow — APPROVED** [review]
LGTM, 2 nits

File: packages/relay/src/lib/precheck.ts [hunk 3]
```diff
@@ -61,7 +81,13 @@ export class Precheck {
     await this.balance(parsedTx, mirrorAccountInfo, requestId);
   }
 
-  async verifyAccount(tx: Transaction, requestId?: string) {
+  /**
+   * Verifies the account.
+   * @param {Transaction} tx - The transaction.
+   * @param {string} [requestId] - The request ID.
+   * @returns {Promise<any>} A Promise.
```
Comment (victor-yanev) [inline packages/relay/src/lib/precheck.ts:88]: Can we also describe the expected object that is resolved from the promise?

File: packages/relay/src/lib/precheck.ts [hunk 4]
```diff
@@ -61,7 +81,13 @@ export class Precheck {
...
+   */
+  async verifyAccount(tx: Transaction, requestId?: string): Promise<any> {
```
Comment (victor-yanev) [inline packages/relay/src/lib/precheck.ts:90]: QQ: Do we know what type of object is expected from `this.mirrorNodeClient.getAccount`? If so, let's change the type from `any` to the expected type.

File: packages/relay/tests/lib/precheck.spec.ts [hunk 5]
```diff
@@ -424,25 +421,20 @@ describe('Precheck', async function () {
...
-    };
-
-    const signed = await signTransaction(defaultTx);
-    const parsedTx = ethers.Transaction.from(signed);
-
-    const mirrorAccount = {
-      evm_address: mockData.accountEvmAddress,
-      ethereum_nonce: defaultNonce,
-    };
+    let defaultNonce, defaultTx, signed, parsedTx, mirrorAccount, wallet;
```
[the remark this answers is not in the document]
Comment (konstantinabl) [reply packages/relay/tests/lib/precheck.spec.ts:424]: Some of them are let cause they are used in the tests below

File: packages/relay/src/lib/precheck.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (konstantinabl) [reply packages/relay/src/lib/precheck.ts]: Sorry, I missed these comments, but have addressed them now

File: packages/relay/tests/lib/precheck.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (victor-yanev) [reply packages/relay/tests/lib/precheck.spec.ts]: It's better now, thanks! I would only move each field to a separate line and change the `defaultNonce` to `const` (it's not changed anywhere I believe?).

**Review by victor-yanev — APPROVED** [review]

**Review by victor-yanev — DISMISSED** [review]

File: packages/relay/tests/lib/eth/eth-helpers.ts [hunk 6]
```diff
@@ -1,3 +1,23 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
```
Comment (georgi-l95) [inline packages/relay/tests/lib/eth/eth-helpers.ts:5]:
```suggestion
 * Copyright (C) 2024 Hedera Hashgraph, LLC
```

**Review by georgi-l95 — DISMISSED** [review]
LG, small nit

**Review by Nana-EC — APPROVED** [review]
LG.
Might need to rebase

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**